modules/tinymce/src/core/main/ts/ForceBlocks.ts

```diff
@@ -43,8 +43,14 @@ const shouldRemoveTextNode = (blockElements: SchemaMap, node: Node) => {
   if (NodeType.isText(node)) {
     if (node.data.length === 0) {
       return true;
-    } else if (/^\s+$/.test(node.data) && (!node.nextSibling || isBlockElement(blockElements, node.nextSibling))) {
-      return true;
+    } else if (/^\s+$/.test(node.data)) {
+      if (!node.nextSibling) {
+        return true;
+      } else if (isBlockElement(blockElements, node.nextSibling)) {
+        return true;
+      } else if (Namespace.isNonHtmlElementRoot(node.nextSibling)) {
+        return true;
+      }
```

Comment:
```suggestion
return !node.nextSibling || isBlockElement(blockElements, node.nextSibling) || Namespace.isNonHtmlElementRoot(node.nextSibling);
```

since they are all true and pretty self explanatory couldn't be better to just return the 3 condition in `or`?

Reply:
Yeah it's hard to know how and when it makes sense to split that long expression out. But you are right that doesn't look to bad to have that expression there as s a one liner.